primefaces/src/main/java/org/primefaces/application/resource/BarcodeHandler.java

```diff
@@ -21,51 +21,62 @@
 import java.util.logging.Logger;
 
 import javax.faces.context.ExternalContext;
 import javax.faces.context.FacesContext;
-import javax.xml.XMLConstants;
-import javax.xml.transform.Result;
-import javax.xml.transform.Source;
-import javax.xml.transform.Transformer;
-import javax.xml.transform.TransformerFactory;
-
-import org.krysalis.barcode4j.HumanReadablePlacement;
-import org.krysalis.barcode4j.output.bitmap.BitmapCanvasProvider;
-import org.krysalis.barcode4j.output.svg.SVGCanvasProvider;
-import org.w3c.dom.DocumentFragment;
+import javax.imageio.ImageIO;
+
+import uk.org.okapibarcode.backend.Codabar;
+import uk.org.okapibarcode.backend.Code128;
+import uk.org.okapibarcode.backend.Code2Of5;
+import uk.org.okapibarcode.backend.Code3Of9;
+import uk.org.okapibarcode.backend.DataMatrix;
+import uk.org.okapibarcode.backend.Ean;
+import uk.org.okapibarcode.backend.HumanReadableLocation;
+import uk.org.okapibarcode.backend.Pdf417;
+import uk.org.okapibarcode.backend.Postnet;
+import uk.org.okapibarcode.backend.QrCode;
+import uk.org.okapibarcode.backend.QrCode.EccLevel;
+import uk.org.okapibarcode.backend.Symbol;
+import uk.org.okapibarcode.backend.Upc;
+import uk.org.okapibarcode.backend.Upc.Mode;
+import uk.org.okapibarcode.graphics.Color;
+import uk.org.okapibarcode.output.Java2DRenderer;
+import uk.org.okapibarcode.output.SvgRenderer;
 
 public class BarcodeHandler extends BaseDynamicContentHandler {
 
     private static final Logger LOGGER = Logger.getLogger(BarcodeHandler.class.getName());
 
-    private final Map<String, BarcodeGenerator> generators;
+    private final Map<String, Symbol> generators;
 
     public BarcodeHandler() {
         generators = new HashMap<>();
-        generators.put("int2of5", new Int2of5Generator());
-        generators.put("codabar", new CodabarGenerator());
-        generators.put("code39", new Code39Generator());
-        generators.put("code128", new Code128Generator());
-        generators.put("ean8", new EAN8Generator());
-        generators.put("ean13", new EAN13Generator());
-        generators.put("upca", new UPCAGenerator());
-        generators.put("upce", new UPCEGenerator());
-        generators.put("postnet", new PostnetGenerator());
-        generators.put("pdf417", new PDF417Generator());
-        generators.put("datamatrix", new DataMatrixGenerator());
+        generators.put("int2of5", new Code2Of5());
+        generators.put("codabar", new Codabar());
+        generators.put("code39", new Code3Of9());
+        generators.put("code128", new Code128());
+        generators.put("ean8", new Ean(Ean.Mode.EAN8));
+        generators.put("ean13", new Ean(Ean.Mode.EAN13));
+        generators.put("upca", new Upc(Mode.UPCA));
+        generators.put("upce", new Upc(Mode.UPCE));
+        generators.put("postnet", new Postnet());
+        generators.put("pdf417", new Pdf417());
+        generators.put("datamatrix", new DataMatrix());
+        generators.put("qr", new QrCode());
```

Comment:
These `Symbol` seems to be stateful (it was certainly the same before) but ideally the state of these objects shouldn't be shared within the same HTTP request (and different requests). This map should act as a factory, so instead of having symbol instance, something like `Pdf417::new`. This way you ensure you have a clean object every time it's requested

Reply:
And +1 to move BarcodeHandler to the .component.barcode package

Reply:
OK yeah i will do the `::new` i saw another library doing that recently so good feedback!

Reply:
Switched to `private final Map<String, Supplier<Symbol>> generators;` so it returns a new instance each time,